runtime/doc/lua.txt

```diff
@@ -1697,14 +1697,14 @@ split({s}, {sep}, {kwargs})                                      *vim.split()*
     Parameters: ~
         {s}       (string) String to split
         {sep}     (string) Separator or pattern
-        {kwargs}  (table) Keyword arguments:
+        {kwargs}  {plain: boolean, trimempty: boolean}|nil Keyword arguments:
```

Comment:
We should probably tweak gen_vimdoc.py to simplify this back to `table`.

Reply:
Or use proper annotation for table keys (`---@field`?) in the docs so we can generate the argument docs as well?

Reply:
@clason for that we would need to add a comment with @class and the correct fields. Is that also how it works in doxygen?

I think for now the current changes are ok and the fields could be a further imrovement?

Reply:
Sure, no need to fix all the issues at once (if they get fixed eventually ;))

`---@class` works, see the treesitter docs, e.g., `query.lua`, but probably  `---@field` is not the right annotation then.

Reply:
With proper lua annotations, we would add something like:

```lua
---@class vim.split.Options
---@field plain boolean some comment
---@field trimempty boolean another comment

---@param opts vim.split.Options options for split...
function split(opts)
```

Reply:
Yeah, that would work! It's a bit verbose in the code, and we'd have to agree to a reasonable convention for the class name, but it's not the worst option by far.

Reply:
Would definitely be very useful for functions that take lots of options, like `nvim_set_extmark` etc...

Reply:
I think that would be great!